packages/web-components/src/tablist/tablist.ts

```diff
@@ -0,0 +1,408 @@
+import { attr, FASTElement, observable } from '@microsoft/fast-element';
+import { wrapInBounds } from '@microsoft/fast-web-utilities';
+import {
+  keyArrowDown,
+  keyArrowLeft,
+  keyArrowRight,
+  keyArrowUp,
+  keyEnd,
+  keyHome,
+  uniqueId,
+} from '@microsoft/fast-web-utilities';
+import { Tab } from '../index.js';
+import { TablistAppearance, TablistOrientation, TablistSize } from './tablist.options.js';
+
+type TabData = Omit<DOMRect, 'top' | 'bottom' | 'left' | 'right' | 'toJSON'>;
+
+/**
+ * A Tablist element that wraps a collection of tab elements
+ * @public
+ */
+export class BaseTablist extends FASTElement {
+  /**
+   * Used for disabling all click and keyboard events for the tabs, child tab elements.
+   * @public
+   * @remarks
+   * HTML Attribute: disabled.
+   */
+  @attr({ mode: 'boolean' })
+  public disabled?: boolean;
+
+  /**
+   * The orientation
+   * @public
+   * @remarks
+   * HTML Attribute: orientation
+   */
+  @attr
+  public orientation: TablistOrientation = TablistOrientation.horizontal;
+  /**
+   * @internal
+   */
+  public orientationChanged(): void {
+    if (this.$fastController.isConnected) {
+      this.setTabs();
+    }
+  }
+  /**
+   * The id of the active tab
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: activeid
+   */
+  @attr
+  public activeid!: string;
+  /**
+   * @internal
+   */
+  public activeidChanged(oldValue: string, newValue: string): void {
+    if (this.$fastController.isConnected && this.tabs.length > 0) {
+      this.prevActiveTabIndex = this.tabs.findIndex((item: HTMLElement) => item.id === oldValue);
+      this.setTabs();
+    }
+  }
+
+  /**
+   * @internal
+   */
+  @observable
+  public tabs!: HTMLElement[];
+  /**
+   * @internal
+   */
+  public tabsChanged(): void {
+    if (this.$fastController.isConnected && this.tabs.length > 0) {
+      this.tabIds = this.getTabIds();
+      this.setTabs();
+    }
+  }
+
+  /**
+   * A reference to the active tab
+   * @public
+   */
+  public activetab!: HTMLElement;
+
+  private prevActiveTabIndex: number = 0;
```

Comment:
Have you tried to store the elements instead of index? The index method can be fragile if an element gets removed or moved.

Reply:
Discussed this issue with @marchbox. tl;dr– because `setTabs` will re-instantiate the tablist on changes to `this.tabs`, the `prevActiveTabIndex` is a fairly shallow mechanism for storing the previous tab's index in the array of tabs when the tabs are click/keyboard interacted. The state is backed up and "saved" in the `<*-tablist activeid>` attribute which also sets `prevActiveTabIndex` on `activeidChanged`, so while brittle it should rescue successfully. 

Resolving this conversation for now.